line-bot-cli/src/main/java/com/linecorp/bot/cli/FriendDemographicsGetCommand.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.bot.cli;
+
+import java.util.Map;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+import org.yaml.snakeyaml.Yaml;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import com.linecorp.bot.client.LineMessagingClient;
+import com.linecorp.bot.model.response.demographics.GetFriendsDemographicsResponse;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Component
+@ConditionalOnProperty(name = "command", havingValue = "friend-demographics-get")
+@AllArgsConstructor(onConstructor = @__(@Autowired))
```

Comment:
Just want to confrim, Is `@Autowired` annotation still required for boot2? IIRC, if a class only has a 1 constructor, spring can inject beans without `@Autowired` or `@Inject` annotation.

Reply:
You are right. It's not required because Spring can auto select collect constructor.

But I like explicit marking and other command do the same way.

https://github.com/line/line-bot-sdk-java/search?utf8=%E2%9C%93&q=%40AllArgsConstructor%28onConstructor+%3D+%40__%28%40Autowired%29%29&type=

How do you think?


p.s. I noticed other module may not use it. On the contrary, some class using field injection instead of constructor's.

Reply:
Okay, let's keep current style and will improve it later.